tests/components/tedee/test_init.py

```diff
@@ -67,3 +67,35 @@ async def test_bridge_device(
     )
     assert device
     assert device == snapshot
+
+
+async def test_cleanup_disconnected_locks(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_tedee: MagicMock,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Ensure disconnected locks are cleaned up."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    devices = dr.async_entries_for_config_entry(
+        device_registry, mock_config_entry.entry_id
+    )
+
+    locks = [device.name for device in devices]
+    assert locks == ["Bridge-AB1C", "Lock-1A2B", "Lock-2C3D"]
+
+    # remove a lock and reload integration
+    mock_tedee.locks_dict.pop(12345)
+
+    await hass.config_entries.async_reload(mock_config_entry.entry_id)
```

Comment:
Can we maybe add a test case where we update the coordinator and check the unavailable state?